test/umschreiben_clj/variables_test.clj

```diff
@@ -258,6 +258,55 @@
                                                     'lower-case
                                                     '[other.namespace :refer [lower-case]]))))))
 
+(deftest handling-duplicate-require
+  (testing "when updating a require and it is already present multiple times"
+    (is (match? (build-code ['[clojure.string :as str]
+                             '[clojure.string :as string]
+                             'clojure.string]
+                            '(do (str/lower-case "a")
+                                 (str/lower-case "c")
+                                 (str/lower-case "d")))
+                (n/sexpr (variables/rename (build-node ['[clojure.string :as str :refer [upper-case]]
+                                                        '[clojure.string :as string :refer [upper-case]]
+                                                        '[clojure.string :refer [upper-case]]]
+                                                       '(do (string/upper-case "a")
+                                                            (upper-case "c")
+                                                            (str/upper-case "d")))
+                                           'clojure.string/upper-case
+                                           'lower-case
+                                           '[clojure.string :as string])))))
```

Comment:
In this case, since the alias that we actually want is present, shouldn't we prefer that one?

It is a small detail and I'm ok going forward without it since this cases are rare, but maybe it is easy to implement.

Reply:
yeah, it is kinda hard. In `umschreiben-clj.variables/rename` you can select the correct ns to use by changing it to:

```clojure
        target-as?        (fn [{:keys [as]}] (= (:as to-require-map) as))
        existing-requires (-> file-node
                             z/edn
                             (#'requires/lookup-require-maps (:namespace to-require-map)))
        existing-require  (or (find-first target-as?
                                          existing-requires)
                              (first existing-requires))
```

but the tricky part is then making the header transformation do the correct modifications. Currently it is "simple" in that it finds the first one, does the specific updates, and then iterates through the others and updates the `:refer` entries accordingly. If we want to make this change we'd need to adapt https://github.com/nubank/umschreiben-clj/pull/2/files#diff-618a78ffc29b1e19598fbd6f2c51936d770528064f1d2ba090c3cfa6bfbad2cdR184 to find the appropriate require and then alter the `:refer` update logic to find all the other entries. It is possible but I couldn't get it running in the last 30 mins so I'm going to leave it for later